### DISCUSSION TITLE
Any way to see future races scheduled for a class?

### DISCUSSION THREAD

**Discussion by EStoakes** [abesto/ams2-career#62] [Ideas]
Hi,

I'm really enjoying this tool and the sense of progression it gives, you guys have done a great job with it.

Just curious if there is any way to see all the races scheduled for a class? I was thinking of setting up a championship rather than just doing single races.

Cheers

**Comment by abesto** [comment]
Glad to hear you're enjoying it :)

At the moment future races don't *exist*: races are generated randomly at the last possible moment. A championship system (and other fun ideas) are being considered over at https://github.com/abesto/ams2-career/discussions/50.

Meanwhile, maybe this helps: under Settings you can enable "Regenerate Races", which gives you a "Regenerate Races" button on the main page. You could use that to turn the app into a random race generator. Tracking the championship will then of course need to happen outside the app, and there's no way to backfill your results for "skipped" races, but maybe it gets you most of what you need?

**Reply by itaska** [reply]
Hi @abesto,

First off, I've been really enjoying your application for two weeks now. I'd like to share some feedbacks as a new user. The experience of current progressing system is a bit 'lossy' because the current race list reflects the current level:

- I liked a certain race configuration yesterday (e.g., beginner kart), but due to the progress of my level by other classes, the one that I liked is gone from the list. I ended up having to race a super kart.

I know the system behaves as designed. I believe you guys already discussed this topic a lot internally. I hope the system is statically cumulative. Once any race is generated randomly, a user should be able to access the race at any moment. I hope you guys consider my feedback as a constructive one. 

Again I really appreciate your efforts.

**Reply by abesto** [reply]
Ooh interesting, I don't think that's something we considered before. We were working with the assumption that your goal is to move up, and never want to go back.

I *think* technically this shouldn't be very hard to implement. Balance wise, lower level events will still generate less XP than higher level events, and higher levels require more XP to move up, so I think that should be fine.

**Reply by itaska** [reply]
But I believe the idea affects your system for the management of race list since the list will keep growing. Anyway thanks for replying.

**Reply by kjdesigns2000** [reply]
HI, this is exactly the idea I came here to discuss. Example, I am a simple hobbyist doing 1-2 races a week/month sometimes. Slower cars and manageable drives are good for me. Now since I keep on progressing I do not see any lower ranked races anymore for the series that interest me. If there is a possibility to get those even after being upgraded to the next level it will be a cool feature to have. Nonetheless the app is excellent and I will keep on using it.

**Reply by abesto** [reply]
Hm, showing *all* series you have access to would quickly make the list too long. That means we'd probably need to add some kind of filtering (probably by grade, discipline; maybe a special "highest grade per discipline"). This is all doable, and even: a good mini-project for someone if they wanted to contribute code!

Meanwhile, here's a thing you can try: on the settings page there's an XP Multiplier box. If you feel like you're moving up to faster cars quicker than you'd like, you can drop that multiplier down. Maybe click the "Long Career" button as a start, see how you like what you see. (FTR: your unlocks / achievements are recalculated on the fly based on race history, you're not losing anything; you can always even stick it back to the default value if you change your mind)
